node/pkg/watchers/evm/connectors/batch_poller.go

```diff
@@ -204,12 +217,13 @@ func (b *BatchPollConnector) getBlocks(ctx context.Context, logger *zap.Logger)
 			return nil, err
 		}
 
+		var n big.Int
 		m := &result.result
 		if m.Number == nil {
-			logger.Error("failed to unmarshal block: Number is nil", zap.Stringer("finality", finality), zap.String("tag", b.batchData[idx].tag))
-			return nil, fmt.Errorf("failed to unmarshal block: Number is nil")
+			logger.Debug("number is nil, treating as zero", zap.Stringer("finality", finality), zap.String("tag", b.batchData[idx].tag))
```

Comment:
should we be publishing `NewBlock`s with zero height?

Reply:
ah, I think I see - this is internal only and this `0` will cause it to skip on line 149